specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-11-01-preview/TuningOptions.json

```diff
@@ -0,0 +1,456 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "PostgreSQLServerManagementClient",
+    "description": "The Microsoft Azure management API provides create, read, update, and delete functionality for Azure PostgreSQL resources including servers, databases, firewall rules, VNET rules, security alert policies, log files and configurations with new business model.",
+    "version": "2024-11-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/flexibleServers/{serverName}/tuningOptions/{tuningOption}": {
+      "get": {
+        "tags": [
+          "TuningOptions"
+        ],
+        "operationId": "TuningOptions_Get",
+        "x-ms-examples": {
+          "TuningOptions_Get": {
+            "$ref": "./examples/Tuning_GetTuningOption.json"
+          }
+        },
+        "description": "Retrieve the tuning option on a server.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v6/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../types/common-types.json#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/TuningOptionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/TuningOptionsResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/flexibleServers/{serverName}/tuningOptions/{tuningOption}/recommendations": {
```

Comment:
>"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/flexibleServers/{serverName}/tuningOptions/{tuningOption}/recommendations" [](http://example.com/codeflow?start=4&length=178)

Also need to add the corresponding singleton endpoint

Reply:
thank you!

Reply:
This is not yet added ? Do not see recommendations/{recommendation} API

Reply:
Bumping this up

Reply:
For this,  recommendations are created automatically and all can be captured in response user. They can filter the recs but there is not current way for getting singleton. 
Also we have gotten this previously approved by ARM with implementation as is.  https://github.com/Azure/azure-rest-api-specs/pull/29295#issuecomment-2162801571
This PR got first approval with TuningOptions.json File June 12th

Reply:
The final approved version of the PR do not have the TuningOptions.json file. 

Per the RPC [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#nested-resources](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#nested-resources) , Nested Resources should also have singleton APIs.

Reply:
From backend team: We would add the singleton option when we will implement the apply action, which would be either a PUT or POST - however this probably is in 6+ months from now

Reply:
@ramoka178 this was a conversation also had with @mentat9, that when the API was first surfaced we got approval by ARM to move forward. These new concerns being brought up are now when we have a time crunch, versus previously when we could have addressed earlier.